app/jobs/dradis/plugins/csv/mapping_import_job.rb

```diff
@@ -0,0 +1,91 @@
+module Dradis::Plugins::CSV
+  class MappingImportJob < ApplicationJob
+    queue_as :dradis_project
+
+    # mappings hash:
+    # The key is the column index, while the value is a hash containing the type of resource (evidence/identifier/issue/node).
+    # It's used to map a CSV header to a field in Dradis (only for evidence and issues).
+    #
+    # e.g.
+    # {
+    #   '0' => { 'type' => 'node' },
+    #   '1' => { 'type' => 'issue', 'field' => 'Title' },
+    #   '2' => { 'type' => 'identifier' },
+    #   '3' => { 'type' => 'evidence', 'field' => 'Port' }
+    # }
+    def perform(file:, mappings:, project_id:, uid:)
+      @logger = Log.new(uid: uid)
+      @logger.write { "Job id is #{job_id}." }
+      @logger.write { 'Worker process starting background task.' }
+
+      # Converts mapping hash into groups of arrays
+      # {
+      #   'node' => [['0', { 'type' => 'node' }]],
+      #   'identifier' => [['2', { 'type' => 'identifier' }]]
+      # }
+      @mappings_groups = mappings.group_by { |index, mapping| mapping['type'] }
+      @id_index = Integer(@mappings_groups['identifier']&.first&.first, exception: false)
+
+      unless @id_index
+        @logger.fatal('Unique Identifier doesn\'t exist, please choose a column as the Unique Identifier.')
+        return
+      end
+
+      @id_index = @id_index.to_i
+      @file = file
+      @project = Project.find(project_id)
+
+      import_csv!
```

Comment:
isn't this what the engine's Importer should do?

The smell was having to define a ContentService instance inside a Job. That's a task solved by `dradis-plugins` and `Dradis::Plugins::Upload::Importer`

Reply:
Moved to the `Importer` class.